app/dojo.py

```diff
@@ -0,0 +1,164 @@
+                print(e)
+
+    @classmethod
+    def print_unallocated(cls, filename):
+        unallocated = Person \
+            .get_unallocated(Staff.get_unallocated_staff(),
+                             Fellow.get_unallocated_fellows())
+        if filename is None:
+            print(unallocated)
+        else:
+            try:
+                new_file = File.create_file(filename)
+                File.write(new_file, unallocated)
+            except FileExistsError as e:
+                print(e)
+
+    @classmethod
+    def reallocate_person(cls, identifier, room_name):
+        person_id = None
+        try:
+            person_id = int(identifier)
+            if Person.exist(person_id):
+                person = Person.id_map[person_id]
+                if Room.exists(room_name):
+                    room = Room.get_room_list()[room_name]
+                    if not room.is_full:
+                        if not (person in room.get_room_members()):
+                            if isinstance(person, Staff):
+                                if room.get_type() == "office":
+                                    Office.reallocate_person(person, room_name)
+                                else:
+                                    print(("Room {} is a livingspace and  "
+                                           + "can't be assigned to a staff")
+                                          .format(room_name))
+                            else:
+                                if room.get_type() == "office":
+                                    Office.reallocate_person(person, room_name)
+                                else:
+                                    LivingSpace.reallocate_person(person,
+                                                                  room_name)
+                        else:
+                            print(("{p.surname} {p.firstname} already belongs "
+                                   + "to room {}, therefore can't be "
+                                   + "reallocated to the same room")
+                                  .format(room_name, p=person))
+                    else:
+                        print("Room {} is filled up, please input another room"
+                              .format(room_name))
+                else:
+                    print("Room {} doesn't exist".format(room_name))
+
+            else:
+                print("Person with the id {} doesn't exist".format(person_id))
+        except ValueError:
+            print("Wrong id format. id must be a number")
+
+    @classmethod
+    def load_people(cls, filename):
+        error_messages = []
+        error_messages.append("Errors\n---------")
+        error_messages.append("The following people couldn't be loaded "
+                              + "because of incomplete information\n")
+        try:
+            file_to_be_loaded = File.open_file(filename)
+            with file_to_be_loaded as data_file:
```

Comment:
i created a list to hold all the error messages that might emanate from adding people via loading a file. The possible errors are incomplete arguments. The list holds error messages for every line in the file that throws an error which i later outputted all together in a well formatted style.
I have refactored line 139 to line 142 in order to eliminate multiple calls to the append method.